File: python/shared/dtos/preset_member_position_dto.py
from . import BaseDTO
from .position_dto import PositionDTO


class PresetMemberPositionDTO(BaseDTO):
    preset_member_position_id: int
    preset_member_id: int
    position_id: int

    model_config = {"from_attributes": True}


class PresetMemberPositionDetailDTO(PresetMemberPositionDTO):
    position: PositionDTO


class AddPresetMemberPositionDTO(BaseDTO):
    position_id: int

File: frontend/src/dtos/presetMemberPositionDto.ts
import type { PositionDTO } from "./positionDto";

export interface PresetMemberPositionDTO {
  presetMemberPositionId: number;
  presetMemberId: number;
  positionId: number;
}

export interface PresetMemberPositionDetailDTO extends PresetMemberPositionDTO {
  position: PositionDTO;
}

export interface AddPresetMemberPositionDTO {
  positionId: number;
}
